### PR TITLE
demo(list): Add accessibility demo page for list

### PR DESCRIPTION
**Pull request by tinayuangao**
PR #6363 is accidentally changed to change of changelog.md

### CODE REVIEW

File: src/lib/list/list.ts [hunk 1]
```diff
@@ -46,35 +46,30 @@ export class MdListDivider {}
...
-export class MdListCssMatStyler {}
-
-/**
- * Directive whose purpose is to add the mat- CSS styling to this selector.
- * @docs-private
- */
-@Directive({
-  selector: 'md-nav-list, mat-nav-list',
-  host: {'class': 'mat-nav-list'}
+  template: '<ng-content></ng-content>',
```
Comment (crisbeto) [inline src/lib/list/list.ts]: I think that in general we still keep the templates in a separate HTML file, even if it's a one-liner.

File: src/demo-app/a11y/list/list-a11y.html [hunk 2]
```diff
@@ -0,0 +1,66 @@
+<div class="demo-list">
+  <section>
+    <h2> Seasoning </h2>
+    <p>Showing a non-interactive list of seasonings.</p>
+    <md-list role="list">
+      <md-list-item *ngFor="let item of items"> {{item}} </md-list-item>
```
Comment (crisbeto) [inline src/demo-app/a11y/list/list-a11y.html]: Shouldn't all of these list items have a `role="listitem"`?

Comment (tinayuangao) [reply src/demo-app/a11y/list/list-a11y.html]: `role="listitem"`  is added in `list.ts` so we don't need to add it here.

Comment (jelbourn) [reply src/demo-app/a11y/list/list-a11y.html]: We're we going to remove that from the list-item?

**Comment by kara** [comment]
@tinayuangao Rebase?

**Pull request merged**